PWGHF/HFC/Tasks/taskCharmHadronsTrackFemtoDream.cxx

```diff
@@ -0,0 +1,1036 @@
+    registryMixQa.add("MixingQA/hMECollisionBins", "; bin; Entries", kTH1F, {{120, -0.5, 119.5}});
+    registryCharmHadronQa.add("CharmHadronQA/hPtVsMass", "; #it{p}_{T} (GeV/#it{c}); inv. mass (GeV/#it{c}^{2})", kTH2F, {binpTCharm, binInvMass});
+
+    if (useCPR.value && process3Prong) {
+      pairCleaner3Prong.init(&registry);
+      pairCloseRejectionSE3Prong.init(&registry, &registry, cprDeltaPhiMax.value, cprDeltaEtaMax.value, cprPlotPerRadii.value, 1);
+      pairCloseRejectionME3Prong.init(&registry, &registry, cprDeltaPhiMax.value, cprDeltaEtaMax.value, cprPlotPerRadii.value, 2);
+    } else if (useCPR.value && process2Prong) {
+      pairCleaner2Prong.init(&registry);
+      pairCloseRejectionSE2Prong.init(&registry, &registry, cprDeltaPhiMax.value, cprDeltaEtaMax.value, cprPlotPerRadii.value, 1);
+      pairCloseRejectionME2Prong.init(&registry, &registry, cprDeltaPhiMax.value, cprDeltaEtaMax.value, cprPlotPerRadii.value, 2);
+    } else if (useCPR.value && processDstar) {
+      pairCleanerDstar.init(&registry);
+      pairCloseRejectionSEDstar.init(&registry, &registry, cprDeltaPhiMax.value, cprDeltaEtaMax.value, cprPlotPerRadii.value, 1);
+      pairCloseRejectionMEDstar.init(&registry, &registry, cprDeltaPhiMax.value, cprDeltaEtaMax.value, cprPlotPerRadii.value, 2);
+    }
+  }
+
+  template <typename CollisionType>
+  void fillCollision(CollisionType const& col)
+  {
+    registryMixQa.fill(HIST("MixingQA/hSECollisionBins"), colBinningMult.getBin({col.posZ(), col.multNtr()}));
+    registryMixQa.fill(HIST("MixingQA/hSECollisionPool"), col.posZ(), col.multNtr());
+  }
+
+  /// Compute the charm hadron candidates mass with the daughter masses
+  /// assumes the candidate is either a D+ or Λc+ or D0 or Dstar
+  template <DecayChannel Channel, typename Candidate>
+  float getCharmHadronMass(const Candidate& cand)
+  {
+    float invMass = 0.0f;
+    if constexpr (Channel == DecayChannel::LcToPKPi) {
+      if (cand.candidateSelFlag() == 1) {
+        invMass = cand.m(std::array{MassProton, MassKPlus, MassPiPlus});
+        return invMass;
+      }
+      invMass = cand.m(std::array{MassPiPlus, MassKPlus, MassProton});
+      return invMass;
+    }
+    // D+ → π K π (PDG: 411)
+    if constexpr (Channel == DecayChannel::DplusToPiKPi) {
+      invMass = cand.m(std::array{MassPiPlus, MassKPlus, MassPiPlus});
+      return invMass;
+    }
+    // D0 → π K  (PDG: 421)
+    if constexpr (Channel == DecayChannel::D0ToPiK) {
+      if (cand.candidateSelFlag() == 1) {
+
+        invMass = cand.m(std::array{MassPiPlus, MassKPlus});
+        return invMass;
+      } else {
+
+        invMass = cand.m(std::array{MassKPlus, MassPiPlus});
+        return invMass;
+      }
+    }
+    // D* → D0π (PDG: 413)
+    if constexpr (Channel == DecayChannel::DstarToD0Pi) {
+      invMass = cand.m(std::array{MassPiPlus, MassKPlus, MassPiPlus});
+      return invMass;
```

Comment:
I would fill the output with the `deltaMass` in case of the `D*` (or at least we should put both `D*` and `D0` masses in the output)